modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/ImageMosaicDatastoreWalker.java

```diff
@@ -233,4 +257,151 @@ public void run() {
             }
         }
     }
+
+    protected void handleCogGranule(final String granuleBeingProcessed) throws IOException {
```

Comment:
I'm guessing this more or less mirrors the handleFile parent method? Any chance this class can be more or less unaware of COGs and just work based on the usage of existing (or new) interfaces encapsulating the different behaviors?

Reply:
I'll take a look but there are at least 2 or 3 different points as well as different types/classes as input being used.